VSCode/armsnippets.json

```diff
@@ -506,6 +506,145 @@
+        "                          \"destinationAddresses\": [",
+        "                              \"[reference(resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')).ipAddress]\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${18|3389,1-64000|}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${19|TCP,UDP,Any,ICMP|}\"",
+        "                          ],",
+        "                          \"translatedAddress\": \"${20:10.0.0.254}\",",
+        "                          \"translatedPort\": \"${21|2500,1-64000|}\"",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"networkRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${22:networkCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${23|100,100-65000|}\",",
+        "                   \"action\": {",
+        "                      \"type\": \"${24|Deny,Allow|}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${25:networkRuleName}\",",
+        "                          \"description\": \"${26:networkRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${27:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${28:10.0.1.4}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${29|1,1-64000,*|}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${30|TCP,UDP,Any,ICMP|}\"",
+        "                          ]",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"ipConfigurations\": [",
+        "           {",
+        "               \"name\": \"${31:ipConfigName}\",",
+        "               \"properties\": {",
+        "                   \"subnet\": {",
+        "                       \"id\": \"${32:[resourceId('virtualNetworkResourceGroup', 'Microsoft.Network/virtualNetworks/subnets', 'virtualNetworkName', 'AzureFirewallSubnet')]}\"",
+        "                   },",
+        "                   \"publicIPAddress\": {",
+        "                       \"id\": \"${33:[resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')]}\"",
```

Comment:
I'd eliminate nested tokens like this:
\\"id\\": \\"[resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')]\\"",

Reply:
That actually wasn't supposed to be there. Good catch.